Design/DesignDoc-Least-ConnectionsTCPLB

```diff
@@ -0,0 +1,181 @@
+# Design document
```

Comment:
nit: If you rename the file and add `.md` extension GitHub will render Markdown correctly including the diagram.